libnavigation-base/src/test/java/com/mapbox/navigation/base/internal/route/AnnotationsRefresherTest.kt

```diff
@@ -0,0 +1,304 @@
+            val defaultAnnotation = LegAnnotation.builder().build()
+            val annotationWithCongestionNumericOnlyEmpty = defaultAnnotation.toBuilder()
+                .congestionNumeric(listOf())
+                .build()
+            val annotationWithCongestionNumericOnlyFilled = defaultAnnotation.toBuilder()
+                .congestionNumeric(listOf(1, 2, 3, 4, 5))
+                .build()
+            return listOf(
+                arrayOf(
+                    null,
+                    null,
+                    0,
+                    null,
+                    "Everything is null"
+                ),
+                arrayOf(
+                    defaultAnnotation,
+                    null,
+                    0,
+                    null,
+                    "New annotation is null"
+                ),
+                arrayOf(
+                    defaultAnnotation,
+                    defaultAnnotation,
+                    0,
+                    defaultAnnotation,
+                    "2 default annotations, index = 0"
+                ),
+                arrayOf(
+                    defaultAnnotation,
+                    defaultAnnotation,
+                    5,
+                    defaultAnnotation,
+                    "2 default annotations, index = 5"
+                ),
+                arrayOf(
+                    annotationWithCongestionNumericOnlyEmpty,
+                    defaultAnnotation,
+                    0,
+                    defaultAnnotation,
+                    "Empty congestion_numeric + default, index = 0"
+                ),
+                arrayOf(
+                    annotationWithCongestionNumericOnlyEmpty,
+                    defaultAnnotation,
+                    0,
+                    defaultAnnotation,
+                    "Empty congestion_numeric + default, index = 5"
+                ),
+                arrayOf(
+                    defaultAnnotation,
+                    annotationWithCongestionNumericOnlyEmpty,
+                    0,
+                    annotationWithCongestionNumericOnlyEmpty,
+                    "Default + empty congestion_numeric, index = 0"
+                ),
+                arrayOf(
+                    defaultAnnotation,
+                    annotationWithCongestionNumericOnlyEmpty,
+                    5,
+                    LegAnnotation.builder().congestionNumeric(listOf(0, 0, 0, 0, 0)).build(),
+                    "Default + empty congestion_numeric, index = 5"
+                ),
+                arrayOf(
+                    annotationWithCongestionNumericOnlyEmpty,
+                    annotationWithCongestionNumericOnlyEmpty,
+                    0,
+                    annotationWithCongestionNumericOnlyEmpty,
+                    "Empty congestion_numeric x2, index = 0",
+                ),
+                arrayOf(
+                    annotationWithCongestionNumericOnlyFilled,
+                    defaultAnnotation,
```

Comment:
asking out of curiosity, can Directions API return empty annotations in route refresh? Is this a regular case or an edge one?

Reply:
Could someone from @mapbox/navigation-api answer above question? 🙏

Reply:
@VysotskiVadim yes, empty annotations can be returned for legs whose index is less than the leg index provided in the refresh request. Eg: for a 2 leg route, when refreshing the **2nd** leg (`.../directions-refresh/v1/mapbox/driving-traffic/<uuid>/0/1`), the first leg's annotations will be empty (`{}`).